_posts/2020-06-18-Metal3-dev-env-BareMetal-Cluster-Deployment.md

```diff
@@ -0,0 +1,396 @@
+| **base** directory | a directory for [`kustomization`](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/kustomization/) resource files, templates of resources created while provisioning               | `${Manifests}/base` |
+| [kustomization](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/kustomization/) file | copied from templates for kustomization in [templates-directory](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test/templates)               | `${Maifests}/kustomization.yaml` |
+
+<br/>
+
+> Note : the `base` directory has templates for the resources that are created while provisioning
+>  - [`kustomize-tool`](https://github.com/kubernetes-sigs/kustomize) is used to build manifests in the base directory
+>  - Cluster, Controlplane and MachineDeployment manifests are generated in `base` directory
+>  - worker manifests are generated and added using kustomize-tool
+>  - All the resources created as part of provisioning have their templates under the ‘base’ directory 
+
+<br/>
+
+
+Here are some of the resources that are created as part of provisioning :
+
+| Name                | Description                                       |
+| ------------------- | ------------------------------------------------- | 
+| Cluster | a Cluster API resource for managing a cluster |
+| Metal3Cluster | Corresponding Metal3 resource generated as part of BareMetal Cluster deployment, and managed by `Cluster`|
+| KubeadmControlPlane | Cluster API resource for managing the Controlplane, it also manages the `Machine` object for Controlplane, and has the **KubeadmConfig** for the Controlplane |
+| MachineDeployment | Cluster API resource for managing workers via `MachineSet` object, it can be used to add/remove workers by scaling Up/Down |
+| MachineSet | Cluster API resource for managing `Machine` objects for worker nodes |
+| Machine | Cluster API resource for managing nodes - Controlplane or Workers. In case of Controlplane, its directly managed by `KubeadmControlPlane`, whereas for Workers it's managed by a `MachineSet`|
+| Metal3Machine | Corresponding Metal3 resource for managing Baremetal Nodes, it's managed by a `Machine` resource |
+| Metal3MachineTemplate | Metal3 resource which acts as a template when creating a Controlplane or a Worker node |
+| KubeadmConfigTemplate | A template of `KubeadmConfig`, for Workers, used to generate KubeadmConfig when a new worker node is provisioned |
+
+> **Note** : The corresponding `KubeadmConfig` is copied to the Controlplane/Worker at the time of provisioning.  
+  
+<br/>
+
+
+## BareMetal Cluster Deployment
+
+The deployment scripts primarily use ansible and the existing Kubernetes management cluster (based on minikube ) for deploying the bare-metal cluster. Make sure that some of the environment variables used for Metal³ deployment are set, if you didn't use `config_${user}.sh` for setting the environment variables.
```

Comment:
We should clarify whether the scripts use ansible for deploying metal3, or only for building the VMs to use for the new cluster. If ansible isn't needed to actually use metal3, we don't want to give anyone the impression that it is.

Reply:
We do use Ansible under the hood to do all the deployments.

Reply:
@dhellmann should I specify it separately that we are using ansible under the hood for all the deployments or does this look okay ?

Reply:
I wasn't sure if the deployment scripts use ansible. Since they do, what you have is clear.